account_invoice_import/wizard/account_invoice_import.py

```diff
@@ -0,0 +1,1044 @@
+            wiz_vals['import_config_id'] = self.import_config_id.id
+            import_config = self.import_config_id.convert_to_import_config()
+        else:  # button called from 'import' step
+            import_configs = aiico.search([
+                ('partner_id', '=', partner.id),
+                ('company_id', '=', company_id)])
+            if not import_configs:
+                raise UserError(_(
+                    "Missing Invoice Import Configuration on partner '%s'.")
+                    % partner.display_name)
+            elif len(import_configs) == 1:
+                wiz_vals['import_config_id'] = import_configs.id
+                import_config = import_configs.convert_to_import_config()
+            else:
+                logger.info(
+                    'There are %d invoice import configs for partner %s',
+                    len(import_configs), partner.display_name)
+
+        if not wiz_vals.get('import_config_id'):
+            wiz_vals['state'] = 'config'
+            action = iaao.for_xml_id(
+                'account_invoice_import',
+                'account_invoice_import_action')
+            action['res_id'] = self.id
+        else:
+            draft_same_supplier_invs = aio.search([
+                ('commercial_partner_id', '=', partner.id),
+                ('type', '=', parsed_inv['type']),
+                ('state', '=', 'draft'),
+                ])
+            logger.debug(
+                'draft_same_supplier_invs=%s', draft_same_supplier_invs)
+            if draft_same_supplier_invs:
+                wiz_vals['state'] = 'update'
+                if len(draft_same_supplier_invs) == 1:
+                    wiz_vals['invoice_id'] = draft_same_supplier_invs[0].id
+                action = iaao.for_xml_id(
+                    'account_invoice_import',
+                    'account_invoice_import_action')
+                action['res_id'] = self.id
+            else:
+                action = self.create_invoice_action(parsed_inv, import_config)
+        self.write(wiz_vals)
+        return action
+
+    @api.multi
+    def create_invoice_action_button(self):
+        '''Workaround for a v10 bug: if I call create_invoice_action()
+        directly from the button, I get the context in parsed_inv'''
+        return self.create_invoice_action()
+
+    @api.multi
+    def create_invoice_action(self, parsed_inv=None, import_config=None):
+        '''parsed_inv is not a required argument'''
+        self.ensure_one()
+        iaao = self.env['ir.actions.act_window']
+        if parsed_inv is None:
+            parsed_inv = self.parse_invoice(
+                self.invoice_file, self.invoice_filename)
+        if import_config is None:
+            assert self.import_config_id
+            import_config = self.import_config_id.convert_to_import_config()
+        invoice = self.create_invoice(parsed_inv, import_config)
+        invoice.message_post(_(
```

Comment:
```suggestion
        invoice.message_post(body=_(
```
Otherwise I get an error.